controllers/depresolver/depresolver_test.go

```diff
@@ -603,15 +603,39 @@ func TestResolveInvalidExtGeoTags(t *testing.T) {
 	}
 }
 
-func TestResolveGeoTagExistsWithinExtGeoTags(t *testing.T) {
+func TestResolveOnlyGeoTagExistsWithinExtGeoTags(t *testing.T) {
 	// arrange
 	defer cleanup()
 	tag := "us-west1"
 	expected := predefinedConfig
 	expected.ClusterGeoTag = tag
-	expected.ExtClustersGeoTags = []string{"us-east1", tag}
-	// act,assert
-	arrangeVariablesAndAssert(t, expected, assert.Error)
+	expected.ExtClustersGeoTags = []string{}
+	configureEnvVar(expected)
+	os.Setenv(ExtClustersGeoTagsKey, tag)
+	resolver := NewDependencyResolver()
+	config, err := resolver.ResolveOperatorConfig()
+	assert.Error(t, err)
+	assert.Equal(t, expected, *config)
+}
+
+func TestResolveGeoTagExistsWithinExtGeoTags(t *testing.T) {
+	// arrange
+	defer cleanup()
+	tag := "us-west1"
+	for _, arr := range [][]string{{"good-tag"}, {"us-east1", "eu"}} {
+		expected := predefinedConfig
+		expected.ClusterGeoTag = tag
+		expected.ExtClustersGeoTags = arr
+		configureEnvVar(expected)
+		os.Setenv(ExtClustersGeoTagsKey, strings.Join(append(arr, tag), ","))
```

Comment:
```go
		expected.ExtClustersGeoTags = arr
		expected.ExtClustersGeoTags = append(expected.ExtClustersGeoTags, tag)
		configureEnvVar(expected)
```
 After resolve, config.ExtClusterGeoTags will contain ExtClusterGeoTag cleaned from ClusterGeoTag - which is value that is expected on this place.
 
 Probably assertion phase needs to be change / extended .

Reply:
I am not sure I get you correctly. In the assert phase we call deep equals on `expected` and `config`.

it's in the loop and
`config. ExtClustersGeoTags` will contain:
1. iter: `{"good-tag"}`
2. iter: `{"us-east1", "eu"}`
however on line 630 we override the os variable `EXT_GSLB_CLUSTERS_GEO_TAGS` to contain whatever is in the `config. ExtClustersGeoTags` + `"us-west1"`
`"us-west1"` is also `== config.ClusterGeoTag` so it should be excluded after config resolution so that `assert.Equal(t, expected, *config)` will hold

Reply:
ok I see, was bit confused by setitng envvar twice 
```go
configureEnvVar(expected)
os.Setenv(ExtClustersGeoTagsKey, strings.Join(append(arr, tag), ","))
```